Modules/ExampleInverter/src/ExampleInverter.cpp

```diff
@@ -0,0 +1,48 @@
+#include "ExampleInverter.h"
+
+#include <mitkImageToItk.h>
+#include <mitkITKImageImport.h>
+#include "mitkLogMacros.h"
+
+namespace captk {
+
+ExampleInverter::ExampleInverter()
+{
+	// any setup required goes here
+}
+
+void ExampleInverter::SetInput(mitk::Image::Pointer inImagePtr)
+{
+	this->m_InputPtr = inImagePtr;
+}
+
+void ExampleInverter::Update()
+{
+	// Read as 3D double image (hardcoded) for now (anything else will crash).
+	// Later we can use predicates and/or AccessByItk to help with this.
+	using TPixel = typename short;
+	const unsigned int VImageDimension = 3;
+	typedef itk::Image<TPixel, VImageDimension> itkImageType;
+	typedef itk::InvertIntensityImageFilter<itkImageType, itkImageType> FilterType;
+
+	// There are a few ways to convert from MITK to ITK...
+	auto itkInImage = mitk::ImageToItkImage<TPixel, VImageDimension>(this->m_InputPtr.GetPointer());
+
+	auto filter = FilterType::New();
+	filter->SetInput(itkInImage);
+	filter->Update();
+	auto itkOutImage = filter->GetOutput();
+
+	// There are also a few ways to convert from ITK to MITK.
+	auto mitkOutImage = mitk::GrabItkImageMemory(itkOutImage); 
+	this->m_LastOutput = mitkOutImage.GetPointer();
```

Comment:
Huh, yet another way to do this! Is there any advantage compared to mitk::ImportItkImage? If yes, please update the slides with the optimal way.

Reply:
I'm not _too_ sure on this, because i find the documentation on this a little lacking, but I think it's a difference in how the memory is managed. I originally had a crash here when using ImportITKImage and had a hunch it was the ITK image memory becoming invalid (presumably due to going out of scope). GrabItkImageMemory definitely causes the mitk::Image to manage the memory and keeps the memory valid for the lifetime of the mitk image. I did find this in the docs:

> If the source (itk image) and the target (mitk image) do not share the same scope, the mitk::GrabItkImageMemory function has to be used instead. Otherwise the image memory managed by the itk image is lost at a scope level change. This affects especially the usage in combination with AccessByItk macros ...

It's possible I'm still missing something of course.

Reply:
To clarify: the crash I received was when using the resulting mitk::Image in the plugin. Not when importing in the module.